test/com/stuartsierra/component_test.clj

```diff
@@ -304,3 +304,57 @@
     (is (= local-key (:dependency-key data)))
     (is (= a (:component data)))
     (is (= system (:system data)))))
+
+(defn protocols-extentable-via-metadata? []
+  (let [{:keys [major minor]} *clojure-version*]
+    (and (= major 1)
```

Comment:
`>=`

Reply:
Fair point!

Here we are using `:extend-via-metadata` in a vanilla way, without relying in any intricacy, so the case that you are concerned about is that of Rich dropping `:extend-via-metadata` altogether.

Seems pretty unlikely :)

Conversely, having your work ceasing to work on 2+ seems like a pain to encounter/debug.